Improvements to enum topic

mairaw opened the pull request:
LiveFyre comment for https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/enumeration-types:

Enum related pages are abusing the enum naming convention. Quite often to see that non-bit flag enum name ended with "s", or its plural form. For example, on this page, the "Days" and "Months" should be named as "Day" and "Month" because there is neither definition make them bit flags nor intention to use them as bit flags. However, the latter example "Days2" is correct since it is named for bit flag enum.

Please use rich diff to see the changes.

mairaw committed:
improvements

mairaw committed:
fixed build errors

rpetrusha commented on docs/csharp/programming-guide/enumeration-types.md:
```diff
@@ -36,77 +29,80 @@ translation.priority.ht:
+
+ When you do not specify values for the elements in the enumerator list, the values are automatically incremented by 1. In the previous example, `Day.Sunday` has a value of 0, `Day.Monday` has a value of 1, and so on. When you create a new `Day` object, it will have a default value of `Day.Sunday` (0) if you do not explicitly assign it a value. When you create an enum, select the most logical def ...
+
+ If the variable `meetingDay` is of type `Day`, then (without an explicit cast) you can only assign it one of the values defined by `Day`. And if the meeting day changes, you can assign a new value from `Days` to `meetingDay`:
```
`Days` --> `Day`

rpetrusha commented on docs/csharp/programming-guide/enumeration-types.md:
```diff
@@ -36,77 +29,80 @@ translation.priority.ht:
+
+ You can use an enumeration type to define bit flags, which enables an instance of the enumeration type to store any combination of the values that are defined in the enumerator list. (Of course, some combinations may not be meaningful or allowed in your program code.)
+
+ You create a bit flags enum by applying the <xref:System.FlagsAttribute?displayProperty=fullName> attribute and defining the values appropriately so that `AND`, `OR`, `NOT` and `XOR` bitwise operations can be performed on them. In a bit flags enum, include a named constant with a value of zero that means "no flags are set." Do not give a flag a value of zero if it does not mean "no flags are set ...
```
This a nit, but personally, I wouldn't fence AND, OR, NOT, or XOR.

mairaw replied:
But then this would open up to be translated. I'm fine either way.

rpetrusha commented on docs/csharp/programming-guide/enumeration-types.md:
```diff
@@ -36,77 +29,80 @@ translation.priority.ht:
+
+ You create a bit flags enum by applying the <xref:System.FlagsAttribute?displayProperty=fullName> attribute and defining the values appropriately so that `AND`, `OR`, `NOT` and `XOR` bitwise operations can be performed on them. In a bit flags enum, include a named constant with a value of zero that means "no flags are set." Do not give a flag a value of zero if it does not mean "no flags are set ...
+
+ In the following example, another version of the `Day` enum, which is named `Days`, is defined. `Days` has the `Flags` attribute and each value is assigned the next greater power of 2. This enables you to create a `Days` variable whose value is `Days.Tuesday` and `Days.Thursday`.
```
attribute and --> attribute, and
`Days.Tuesday` and `Days.Thursday` --> It's unclear what the "and" means here, and confusing because it's not a Boolean Boolean. Perhaps `Days.Tuesday | Days.Thursday`?

rpetrusha reviewed (changes requested):
@mairaw, a couple of comments, though this looks good.

mairaw committed:
feedback

mairaw commented:
@rpetrusha thanks for your comments. I've addressed your feedback with the exception of the fenced operations, which I wasn't sure whether we want to do that or not. This is ready for a new review.

rpetrusha commented:
@mairaw, I think that this looks good. Unless you want @BillWagner to review it first, it's ready to merge.
